analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt

```diff
@@ -43,296 +56,278 @@ import org.ossreviewtoolkit.model.Scope
+            // Cache responses more aggressively.
+            addNetworkInterceptor { chain ->
+                val response = chain.proceed(chain.request())
+                val cacheControl = CacheControl.Builder()
+                    .maxAge(7, TimeUnit.DAYS)
+                    .build()
+                response.newBuilder()
+                    .header("cache-control", cacheControl.toString())
+                    .build()
+            }
+        }.use { response ->
+            if (response.cacheResponse != null) {
+                log.debug { "Retrieved '$url' response from local cache." }
+            } else {
+                log.debug { "Retrieved '$url' response from remote server." }
+            }
 
-    private fun jsonNodeToPackage(packageContent: Pair<String, JsonNode>): Package {
-        val (nuspecUrl, jsonCatalogNode) = packageContent
-        val jsonNuspecNode = try {
-            xmlMapper.readTree(
-                nuspecUrl.replace(
-                    "${jsonCatalogNode["version"].textValueOrEmpty()}.nupkg",
-                    "nuspec"
-                ).requestFromNugetAPI()
-            )
-        } catch (e: IOException) {
-            EMPTY_JSON_NODE
+            response.body?.string()?.takeIf { response.isSuccessful }
+                ?: throw IOException("Failed to get a response body from '$url'.")
         }
 
-        if (jsonCatalogNode["id"]?.textValue() == null) return Package.EMPTY
-
-        val vcsInfo = extractVcsInfo(jsonNuspecNode["metadata"] ?: EMPTY_JSON_NODE)
-
-        return Package(
-            id = extractPackageId(jsonCatalogNode),
-            declaredLicenses = extractDeclaredLicenses(jsonCatalogNode),
-            description = jsonCatalogNode["description"].textValueOrEmpty(),
-            homepageUrl = jsonCatalogNode["projectUrl"].textValueOrEmpty(),
-            binaryArtifact = extractRemoteArtifact(jsonCatalogNode, packageContent.first),
-            sourceArtifact = RemoteArtifact.EMPTY,
-            vcs = vcsInfo
-        )
+        return mapper.readValue(body)
     }
 
-    private fun getPackageReferenceFromRestAPI(packageID: String, version: String): PackageReference? {
-        val packageJsonNode = preparePackageReference(packageID, version)
+    private fun getAllPackageData(id: Identifier): AllPackageData {
+        // Note: The package name in the URL is case-sensitive and must be lower-case!
+        val lowerId = id.name.toLowerCase()
+        val dataUrl = "$registrationsBaseUrl/$lowerId/${id.version}.json"
+        val data = mapFromUrl<PackageData>(jsonMapper, dataUrl)
 
-        if (packageJsonNode == null) {
-            issues += createAndLogIssue(
-                source = "nuget-API",
-                message = "$packageID:$version can not be found on Nugets RestAPI."
-            )
+        val nupkgUrl = data.packageContent
+        val nuspecUrl = nupkgUrl.replace(".${id.version}.nupkg", ".nuspec")
 
-            return null
+        return runBlocking {
```

Comment:
Is `runBlocking` actually needed? (e.g. could it be just `run` instead?

I might be missing some basics here.

Reply:
`run` is something completely different (similar to `let`) and not related to coroutines at all.

Reply:
Hm, mind giving me a hint why `runBlocking` is used?

Reply:
Do you mean why `runBlocking` is needed at all, or why I'm using it instead of something else? I assume the former. So, it's needed to bridge suspend function to regular code, also see https://kotlinlang.org/docs/tutorials/coroutines/coroutines-basic-jvm.html#async-returning-a-value-from-a-coroutine and https://kotlinlang.org/docs/reference/coroutines/basics.html#bridging-blocking-and-non-blocking-worlds.